libs/ui/src/components/activity-section-list/activity-section-list.tsx

```diff
@@ -45,16 +47,47 @@ const renderSectionHeader = (item: { section: SectionListData<ActivityData, Sect
 
 const emptyIconSize = getCustomSize(isWeb ? 30 : 36);
 
-export const ActivitySectionList: FC<Props> = ({ publicKeyHash, chainId, tokenAddress = '' }) => {
+export const ActivitySectionList: FC<Props> = ({ publicKeyHash, chainId, filterType, tokenAddress = '' }) => {
   const [offsetY, setOffsetY] = useState(0);
-  const { activity, fetch, isLoading } = useAllActivity(publicKeyHash, getDebankId(chainId), tokenAddress);
+  const { activity: allActivity, fetch, isLoading } = useAllActivity(publicKeyHash, getDebankId(chainId), tokenAddress);
+
+  const [activity, setActivity] = useState(allActivity);
 
   const handleFetchData = () => {
     if (offsetY === 0) {
       fetch(0);
     }
   };
 
+  const filterActivity = (condition: (activity: ActivityData) => boolean) => {
+    setActivity(
+      allActivity
+        .map(item => ({
+          title: item.title,
+          data: item.data.filter(activity => condition(activity))
+        }))
+        .filter(item => item.data.length)
+    );
+  };
+
+  useEffect(() => {
+    if (isDefined(filterType)) {
+      if (filterType === TransactionTypeEnum.Send) {
+        filterActivity(activity => activity.type === TransactionTypeEnum.Send);
+      } else if (filterType === TransactionTypeEnum.Receive) {
+        filterActivity(activity => activity.type === TransactionTypeEnum.Receive);
+      } else if (filterType === TransactionTypeEnum.ContractInteraction) {
+        filterActivity(activity => activity.type === TransactionTypeEnum.ContractInteraction);
+      } else if (filterType === 'collectibles') {
```

Comment:
mb better to rewrite it to switch.. case construction ?

Reply:
idk, seem good :)